src/openrct2/ride/ride_group.c

```diff
@@ -0,0 +1,221 @@
+        else
+            return &ride_group_hypercoaster;
+    case RIDE_TYPE_JUNIOR_ROLLER_COASTER:
+        if (ride_entry_get_supported_track_pieces(rideEntry) & (1ULL << TRACK_SLOPE_STEEP))
+            return &ride_group_midi_coaster;
+        else
+            return &ride_group_junior_rc;
+    case RIDE_TYPE_CAR_RIDE:
+        if (rideEntry->enabledTrackPieces & (1ULL << TRACK_SLOPE_STEEP))
+            return &ride_group_monster_trucks;
+        else
+            return &ride_group_car_ride;
+    case RIDE_TYPE_TWISTER_ROLLER_COASTER:
+        if (rideEntry->enabledTrackPieces & (1ULL << TRACK_VERTICAL_LOOP))
+            return &ride_group_steel_twister_rc;
+        else
+            return &ride_group_hyper_twister;
+    default:
+        // Fall back to the track defaults.
+        return ride_group_generate_fallback(trackType);
+    }
+}
+
+bool track_type_has_ride_groups(uint8 trackType)
+{
+    if (!gConfigInterface.select_by_track_type) {
+        return false;
+    }
+    
+    switch (trackType)
+    {
+    case RIDE_TYPE_CORKSCREW_ROLLER_COASTER:
+    case RIDE_TYPE_JUNIOR_ROLLER_COASTER:
+    case RIDE_TYPE_CAR_RIDE:
+    case RIDE_TYPE_TWISTER_ROLLER_COASTER:
+        return true;
+    default:
+        return false;
+    }
+}
+
+void generate_ride_groups()
+{
+    corkscrew_rc_groups[0] = ride_group_corkscrew_rc;
+    corkscrew_rc_groups[1] = ride_group_hypercoaster;
+    
+    junior_rc_groups[0] = ride_group_junior_rc;
+    junior_rc_groups[1] = ride_group_midi_coaster;
+    
+    car_ride_groups[0] = ride_group_car_ride;
+    car_ride_groups[1] = ride_group_monster_trucks;
+    
+    twister_rc_groups[0] = ride_group_steel_twister_rc;
+    twister_rc_groups[1] = ride_group_hyper_twister;
+}
+
+ride_group * ride_group_generate_fallback(uint8 trackType)
+{
+    fallback_group.track_type = trackType;
+    fallback_group.maximum_height = RideData5[trackType].max_height;
+    fallback_group.available_track_pieces = RideTypePossibleTrackConfigurations[trackType];
+    fallback_group.naming = RideNaming[trackType];
+
+    return &fallback_group;
+}
+
+ride_group * ride_group_find(uint8 rideType, uint8 index)
+{
+    ride_group * rideGroup;
+    generate_ride_groups();
+    
+    switch(rideType)
+    {
+        case RIDE_TYPE_CORKSCREW_ROLLER_COASTER:
+            rideGroup = &corkscrew_rc_groups[index];
+            break;
+        case RIDE_TYPE_JUNIOR_ROLLER_COASTER:
+            rideGroup = &junior_rc_groups[index];
+            break;
+        case RIDE_TYPE_CAR_RIDE:
+            rideGroup = &car_ride_groups[index];
+            break;
+        case RIDE_TYPE_TWISTER_ROLLER_COASTER:
+            rideGroup = &twister_rc_groups[index];
+            break;
+        default:
+            return NULL;
+    }
+    
+    return rideGroup;
+}
+
+bool ride_groups_are_equal(ride_group * a, ride_group * b)
+{
+    if (a->naming.name == b->naming.name && a->naming.description == b->naming.description)
```

Comment:
Why not just `return (a->naming.name == b->naming.name && a->naming.description == b->naming.description)`?